source/slang/ir.cpp

```diff
@@ -4526,13 +4607,16 @@ namespace Slang
             specMangledName = getMangledName(specDeclRef);
         else
             specMangledName = mangleSpecializedFuncName(genericFunc->mangledName, specDeclRef.substitutions);
-
+        RefPtr<IRSpecSymbol> symb;
+        if (sharedContext->symbols.TryGetValue(specMangledName, symb))
```

Comment:
I just flat out don't get why sometimes we use the `symbols` dictionary to find an existing value with the given name, and sometimes we use a simple loop over all global values.